internal/api/employee.go

```diff
@@ -0,0 +1,164 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/SharanyaSD/Payroll-GoLang.git/internal/app/emp"
+	"github.com/SharanyaSD/Payroll-GoLang.git/internal/pkg/dto"
+)
+
+func CreateEmployeeHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var req dto.CreateEmployeeRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		fmt.Printf("%+v", req)
+
+		if req.DateOfBirth.After(time.Now()) {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Date of birth cannot be in the future"))
+			return
+		}
+
+		employeeInfo, err := empSvc.CreateEmployee(req)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		responseJSON, err := json.Marshal(employeeInfo)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+	}
+}
+
+func DeleteEmployeeHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		empID := r.URL.Query().Get("id")
+		if empID == "" {
+			http.Error(w, "Employee ID is required", http.StatusBadRequest)
+			return
+		}
+
+		deletedEmp, err := empSvc.DeleteEmployee(empID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		jsonData, err := json.Marshal(deletedEmp)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("Employee deleted successfully"))
+		w.Write(jsonData)
+	}
+}
+
+func GetAllEmployeesHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		response, err := empSvc.GetAllEmployees()
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		responseJSON, err := json.Marshal(response)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+
+	}
+}
+
+func GetEmployeeByIDHandler(empSvc emp.Service) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		id := r.URL.Query().Get("id")
+
+		if id == "" {
+			http.Error(w, "No ID provided", http.StatusBadRequest)
+			return
+		}
+
+		employee, err := empSvc.GetEmployeeByID(id)
+		if err != nil {
+			// w.WriteHeader(http.StatusInternalServerError)
+			// w.Write([]byte(err.Error()))
+			// return
+			http.Error(w, "Failed to get employee: "+err.Error(), http.StatusNotFound)
```

Comment:
incorrect statu code, it shoud be internal server error onlu.

add logic to check if the employee object has any data and then return StatusNotFound